src/App.tsx

```diff
@@ -16,35 +18,94 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  let goods = [...goodsFromServer];
+  let isReverseCLicked = false;
+  const ALPHABETICALLY: string = 'alphabetically';
+  const BY_LENGTH: string = 'by length';
+  const RESET: string = 'reset';
+  const [sortedGoods, setSortedGoods] = useState('');
+  const [reverse, setReversed] = useState(false);
+
+  function sortGoods() {
+    if (sortedGoods === ALPHABETICALLY) {
+      goods = goods.toSorted((goodA, goodB) => goodA.localeCompare(goodB));
+    }
+
+    if (sortedGoods === BY_LENGTH) {
+      goods = goods.toSorted((goodA, goodB) => goodA.length - goodB.length);
+    }
+
+    if (reverse) {
+      goods = goods.toReversed();
+    }
+  }
```

Comment:
[CODE STYLE] - Function's name should start with a verb (so you could easily tell what your method actually do). Consider renaming `sortGoods` to `applySortingAndReversing` or something similar that reflects all the actions performed by the function.